compiler/desugared/from_surface.ml

```diff
@@ -528,13 +528,22 @@ let rec translate_expr
         StructField.Map.empty fields
     in
     let expected_s_fields = StructName.Map.find s_uid ctxt.structs in
-    StructField.Map.iter
-      (fun expected_f _ ->
-        if not (StructField.Map.mem expected_f s_fields) then
-          Message.raise_spanned_error pos
-            "Missing field for structure %a: \"%a\"" StructName.format s_uid
-            StructField.format expected_f)
-      expected_s_fields;
+    if
+      StructField.Map.exists
+        (fun expected_f _ -> not (StructField.Map.mem expected_f s_fields))
+        expected_s_fields
```

Comment:
I generally find it nicer to write `let missing = Map.diff expected fields in if not (Map.is_empty missing) then ...`, but it's generally less efficient and well `Map` doesn't have a proper `diff` function (although you can use `merge` for that :shrug:) so this is fine ;)

Reply:
I didn't know you coud use `Map.merge` to implement some sort of `Map.diff`. I'll remember it next time :)